src/geocad/src/TGeoToOCC.cxx

```diff
@@ -388,16 +388,20 @@ TopoDS_Shape TGeoToOCC::OCC_Sphere(Double_t rmin, Double_t rmax,
    TopoDS_Wire w;
 
 
-   if(rmin==0&&phi1==0&&Dphi==2*M_PI&&theta1==0&&Dtheta==M_PI) {
-      TopoDS_Solid s= BRepPrimAPI_MakeSphere(rmax);
+   if (rmin < Precision::Confusion() &&
+       phi1 < Precision::Confusion() &&
+       fabs(Dphi - 2*M_PI) < Precision::Confusion() &&
+       theta1 < Precision::Confusion() &&
```

Comment:
```suggestion
   if (rmin < Precision::Confusion() &&
       phi1 < Precision::Confusion() &&
       fabs(Dphi - 2*M_PI) < Precision::Confusion() &&
       theta1 < Precision::Confusion() &&
```
Missing fabs at least for the `phi1`?

Reply:
Or `2 * M_PI - D_phi < Precision::Confusion()`.

Reply:
```suggestion
   if (rmin < Precision::Confusion() &&
       fabs(phi1) < Precision::Confusion() &&
       fabs(Dphi - 2*M_PI) < Precision::Confusion() &&
       theta1 < Precision::Confusion() &&
```

Reply:
> Or `2 * M_PI - D_phi < Precision::Confusion()`.

Not sure what you mean here. I added the fabs on phi1, kept the fabs on Dphi.

Reply:
The assumption might be that 0 <= phi1 <= 2 * M_PI, then fabs expands with a minus sign.